overlays/holo-nixpkgs/hpos-holochain-api/src/index.js

```diff
@@ -57,28 +57,38 @@ const getPresentedHapps = async usageTimeInterval => {
   return presentedHapps
 }
 
-app.get('/hosted_happs', async (req, res) => {
-  const usageTimeInterval = await Promise.race([
+// process the 'body' of the request, and parse it into JSON
+// falling back to returning undefined either if no body is passed
+// or the string fails to parse as valid JSON
+const handleBodyData = (req) => {
+  return Promise.race([
     new Promise(resolve => req.on('data', (body) => {
-      resolve(JSON.parse(body.toString()))
+      try {
+        resolve(JSON.parse(body.toString()))
+      } catch (_) {
+        // parse error
+        resolve(undefined)
+      }
```

Comment:
Can we log an error here?
```suggestion
      } catch (err) {
        console.error("Failed to parse request body:")
        console.log(err)
        resolve(undefined)
      }
```